src/main/java/com/splunk/hecclient/HttpClientBuilder.java

```diff
@@ -74,6 +74,9 @@ public CloseableHttpClient build() {
                 .setSoTimeout(socketTimeout * 1000)
                 .build();
         RequestConfig requestConfig = RequestConfig.custom()
+                .setSocketTimeout(socketTimeout)
+                .setConnectionRequestTimeout(socketTimeout)
+                .setConnectTimeout(socketTimeout)
```

Comment:
I checked and `splunk.hec.socket.timeout` is in sec - thus we probably need to multiply by 1000 to ensure that units are correct. Could you please double check if it is required for each of these parameters settings. 
```suggestion
                .setSocketTimeout(socketTimeout*1000)
                .setConnectionRequestTimeout(socketTimeout*1000)
                .setConnectTimeout(socketTimeout*1000)
```

Reply:
Also another question is - do we want to have different configs or let sockettimeout control other parameters as well (like ConnectionRequestTimeout and ConnectTimeout)?

Reply:
Added connectionRequestTimeout and connectionTimeout as params